cdap-common/src/main/java/co/cask/cdap/common/exception/HttpExceptionHandler.java

```diff
@@ -0,0 +1,60 @@
+/*
+ * Copyright © 2015 Cask Data, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not
+ * use this file except in compliance with the License. You may obtain a copy of
+ * the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations under
+ * the License.
+ */
+
+package co.cask.cdap.common.exception;
+
+import co.cask.cdap.common.http.SecurityRequestContext;
+import co.cask.http.ExceptionHandler;
+import co.cask.http.HttpResponder;
+import org.jboss.netty.handler.codec.http.HttpRequest;
+import org.jboss.netty.handler.codec.http.HttpResponseStatus;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/**
+ * Common logic to handle exceptions in handler methods.
+ */
+public class HttpExceptionHandler extends ExceptionHandler {
+
+  private static final Logger LOG = LoggerFactory.getLogger(HttpExceptionHandler.class);
+
+  @Override
+  public void handle(Throwable t, HttpRequest request, HttpResponder responder) {
+    if (t instanceof IllegalArgumentException) {
```

Comment:
I can guess that we mean to handle the exceptions thrown from ``Preconditions.checkArgument`` here. It seems a little too generic, but unsure how we can get around to changing the ``IllegalArgumentException`` to a ``BadRequestException`` without adding a fair bit of boiler-plate code.